beacon-chain/blockchain/process_block.go

```diff
@@ -125,6 +125,19 @@ func (s *Service) onBlock(ctx context.Context, signed *ethpb.SignedBeaconBlock,
 			return err
 		}
 	}
+	if featureconfig.Get().UpdateHeadTimely {
+		if postState.FinalizedCheckpointEpoch() > s.finalizedCheckpt.Epoch {
```

Comment:
So with triggering this over here, you end up missing forkchoice pruning, deposit trie pruning etc. As the finalized checkpoints are updated here and will not be triggered below again.